app/src/renderer/components/staking/DelegationModal.vue

```diff
@@ -142,80 +105,70 @@ export default {
   },
   data: () => ({
     amount: null,
-    password: ``,
-    selectedIndex: 0,
-    sending: false
+    selectedIndex: 0
   }),
   computed: {
-    ...mapGetters([`wallet`, `connected`, `delegates`]),
+    ...mapGetters([`wallet`, `delegates`]),
+    balance() {
+      return this.fromOptions[this.selectedIndex].maximum
+    },
     from() {
       return this.fromOptions[this.selectedIndex].address
     }
   },
   methods: {
-    close() {
-      this.$emit(`update:showDelegationModal`, false)
+    open() {
+      this.$refs.actionModal.open()
     },
-    async validateForm() {
+    validateForm() {
       this.$v.$touch()
 
-      if (!this.$v.$invalid) {
-        await this.submitForm()
-      }
+      return !this.$v.$invalid
     },
-    async submitDelegation() {
-      await this.$refs.actionModal.submit(async () => {
-        await this.$store.dispatch(`submitDelegation`, {
-          validator_addr: this.validator.operator_address,
-          amount: String(this.amount),
-          password: this.password
-        })
+    async submitDelegation(submitType, password) {
+      await this.$store.dispatch(`submitDelegation`, {
+        validator_addr: this.validator.operator_address,
+        amount: String(this.amount),
+        submitType,
+        password
+      })
 
-        this.$store.commit(`notify`, {
-          title: `Successful delegation!`,
-          body: `You have successfully delegated your ${this.denom}s`
-        })
-      }, `Submitting delegation failed`)
+      this.$store.commit(`notify`, {
+        title: `Successful delegation!`,
+        body: `You have successfully delegated your ${this.denom}s`
+      })
     },
-    async submitRedelegation() {
-      await this.$refs.actionModal.submit(async () => {
-        const validatorSrc = this.delegates.delegates.find(
-          v => this.from === v.operator_address
-        )
-        await this.$store.dispatch(`submitRedelegation`, {
-          validatorSrc,
-          validatorDst: this.validator,
-          amount: String(this.amount),
-          password: this.password
-        })
+    async submitRedelegation(submitType, password) {
+      const validatorSrc = this.delegates.delegates.find(
```

Comment:
we should make `delegates.delegates` an object: `{address: validatorObject}` (not this PR but I'll create an issue for a modules refactor)